docs/ValidationRules.md

```diff
@@ -0,0 +1,642 @@
+# Rules
+
+## Rule `BA3001.EnablePositionIndependentExecutable`
+
+### Description
+
+A Position Independent Executable (PIE) relocates all of its sections at load time, including the code section, if ASLR is enabled in the Linux kernel (instead of just the stack/heap).  This makes ROP-style attacks more difficult. This can be enabled by passing '-f pie' to clang/gcc.
+
+### Messages
+
+#### `Pass_Executable`: Error
```

Comment:
>Error [](start = 24, length = 5)

this is a 'pass' message, not an error. i notice all the messages are emitted as 'Error'.

Reply:
Fixed!